Hakawai/Mentions/HKWMentionsPlugin.m

```diff
@@ -846,6 +853,52 @@ - (void)toggleAutocorrectAsRequiredForRange:(NSRange)range {
     }
 }
 
+- (void)highlightMentionIfNeededForCursorLocation:(NSUInteger)cursorLocation {
+    // When cursor is placed within mention, highlight that mention.
+    __strong __auto_type parentTextView = self.parentTextView;
+    const NSRange textFullRange = HKW_FULL_RANGE(parentTextView.attributedText);
+
+    // When whitespace is inserted at end of text, it is not included in text range.
+    // So check if cursor location is inside text range.
+    if (NSLocationInRange(cursorLocation, textFullRange)) {
+        const NSRange currentMentionRange = [HKWMentionsPlugin mentionRangeAtLocation:cursorLocation attributedText:parentTextView.attributedText];
```

Comment:
https://developer.apple.com/documentation/foundation/nsattributedstring/1408174-attribute

I think you can use this. HKWMentionsAttribute is the attribute. And then you don't need to have the extra function! And then you can do stuff based on the equal range == currentlySelectedMentionRangeV2?

Reply:
We have tried to implement `attribute(at:effectiveRange:)` but this method has issue when we move over normal text.
`attribute(at:effectiveRange:)` return mention range for normal text as well.
So we have used similar method (i.e `enumerateAttributesInRange(:options:usingBlock)` which also iterates over attributes) and not considers normal text as mention text.